### ISSUE TITLE
DOC: Document START_SELECTION_WEIGHTS tuning methodology and training/test set hygiene

### ISSUE THREAD

**Issue by roeimed0** [roeimed0/bacterial-gene-prediction#51]
## What documentation is affected?

- [x] Inline comments

## Location

`src/config.py` — `START_SELECTION_WEIGHTS` (lines 73–79)

## What is wrong or missing?

`START_SELECTION_WEIGHTS` contains specific non-round values:
```python
START_SELECTION_WEIGHTS = {
    'codon': 4.8562,
    'imm': 1.0107,
    'rbs': 0.6383,
    'length': 7.4367,
    'start': 0.2755
}
```

No comment explains:
- How these values were derived (optimization method, objective function)
- Which genomes were used for tuning
- **Whether `TEST_GENOMES` were excluded** — this is a data contamination question
- What the F1 was before vs after optimization

## Suggested improvement

Add a comment block before the dict:

```python
# START_SELECTION_WEIGHTS — tuned via scipy.optimize.differential_evolution
# Objective: maximize F1 for start site selection on training set
# Training set: [list 27 genome accessions here]
# TEST_GENOMES (src/config.py lines 12-31): EXCLUDED from tuning
# F1 before optimization: X.XXX | F1 after: X.XXX
# Method: 100 iterations, population=15, tolerance=1e-6
# Date tuned: [date]
```

Also: add an assertion in `tests/test_config.py` (TST issue #33) that verifies no `TEST_GENOMES` accession appears in the documented training set, preventing accidental contamination in future tuning runs.

## Additional context

Per `role_ml_engineer.md` (line 248): "Never tune thresholds on the benchmark set — that is test set contamination." This issue creates the audit trail proving hygiene was maintained.

**Comment by roeimed0** [comment]
Covered by models/MODEL_LOG.md and models/calibration_report.json. The weights were calibrated (scripts/calibrate_start_weights.py) and confirmed optimal via coordinate sensitivity analysis on 100 genomes.
